src/Common/src/Interop/OSX/Interop.RunLoop.cs

```diff
@@ -26,7 +27,11 @@ internal static partial class RunLoop
         /// <summary>
         /// Starts the current thread's RunLoop. If the RunLoop is already running, creates a new, nested, RunLoop in the same stack.
         /// </summary>
+#if MONO
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+#else
         [DllImport(Interop.Libraries.CoreFoundationLibrary)]
+#endif
```

Comment:
I should have clarified that this is not an import of all the FSW code, only the parts needed for the OSX one. There will be more friction when we are importing all of them and having them not fill in the same partial class.

There may be a case for a light "Mono transform" script that e.g. renames and duplicates the 'general' partial class files we end up using, so that full renamed classes are available via a shim. We also talked about putting sanity checks to prevent invoking interop code for the wrong platform. Regarding the PAL/System.Native.dylib, we are trying building it as part of Mono instead of extracting code.